pipenv/core.py

```diff
@@ -507,12 +497,15 @@ def ensure_project(
 
     # Clear the caches, if appropriate.
     if clear:
+        print("here")
         print("clearing")
         sys.exit(1)
 
     # Automatically use an activated virtualenv.
-    if project.s.PIPENV_USE_SYSTEM:
-        system = True
+    # checks if virtual Enviroment exists if it exists then it makes the system variable true
+    if project.virtualenv_exists:
```

Comment:
(2) This should probably be an or now:
`if project.s.PIPENV_USE_SYSTEM or project.virtualenv_exists:`